e2e/test/scenarios/native-filters/reproductions/31606.cy.spec.js

```diff
@@ -0,0 +1,116 @@
+import {
+  filterWidget,
+  openNativeEditor,
+  restore,
+  queryBuilderMain,
+  popover,
+} from "e2e/support/helpers";
+import * as SQLFilter from "../helpers/e2e-sql-filter-helpers";
+import * as FieldFilter from "../helpers/e2e-field-filter-helpers";
+
+const SQL_QUERY = "SELECT * FROM PRODUCTS WHERE CATEGORY = {{test}}";
+
+describe("issue 31606", { tags: "@external" }, () => {
+  beforeEach(() => {
+    cy.intercept("POST", "/api/dataset").as("dataset");
+
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should clear values on UI for Text, Number, Date and Field Filter Types (metabase#31606)", () => {
+    openNativeEditor();
+
+    SQLFilter.enterParameterizedQuery(SQL_QUERY);
+
+    // Text
+    SQLFilter.setWidgetValue("Gizmo");
+    SQLFilter.runQuery();
+
+    queryBuilderMain()
+      .findByText(/missing required parameters/)
+      .should("not.exist");
+
+    filterWidget().findByRole("textbox").clear();
+
+    SQLFilter.runQuery();
+    queryBuilderMain()
+      .findByText(/missing required parameters/)
+      .should("be.visible");
+
+    filterWidget().within(() => {
+      cy.icon("close").should("not.exist");
+    });
+
+    SQLFilter.openTypePickerFromDefaultFilterType();
+    SQLFilter.chooseType("Number");
+    SQLFilter.setWidgetValue("123");
+
+    SQLFilter.runQuery();
+
+    queryBuilderMain()
+      .findByText(/missing required parameters/)
+      .should("not.exist");
+
+    filterWidget().findByRole("textbox").clear();
+    SQLFilter.runQuery();
+    queryBuilderMain()
+      .findByText(/missing required parameters/)
+      .should("be.visible");
+    filterWidget().within(() => {
+      cy.icon("close").should("not.exist");
+    });
```

Comment:
We leave lots of comments regarding this notation, perhaps we should discuss it during frontend guild meeting and discuss what to do with it.

The alternative being of course:

```suggestion
    filterWidget().icon("close").should("not.exist");
```

No need to change it, I'm just pointing out that we should discuss it.

Reply:
right, thanks. you can leave a comment like `nit:`, it means you allow person to skip the comment if it doesn't work for him 😄 

you're right, I just was confused by `cy.icon`, I thought it would work only directly on `cy` object and it's not chainable.